pytorch_lightning/trainer/trainer.py

```diff
@@ -2022,19 +2017,26 @@ def optimizers(self, new_optims: Optional[List[Optimizer]]) -> None:
         self.strategy.optimizers = new_optims
 
     @property
-    def lr_schedulers(self) -> List[LRSchedulerConfig]:
+    def lr_scheduler_configs(self) -> List[LRSchedulerConfig]:
         return self.strategy.lr_schedulers
 
-    @lr_schedulers.setter
-    def lr_schedulers(self, new_schedulers: List[LRSchedulerConfig]) -> None:
-        self.strategy.lr_schedulers = new_schedulers
+    @property
+    def lr_schedulers(self) -> List[Dict[str, Any]]:
+        rank_zero_deprecation(
+            "`Trainer.lr_schedulers` is deprecated in v1.6 and will be removed in v1.8."
+            " You can use `trainer.lr_scheduler_configs` instead which contains dataclasses instead of dictionaries.",
```

Comment:
do we need a way for people to access schedulers directly outside the LightningModule? If so, I strongly recommend to change the deprecation to just a warning that mentions that the output changes to the list of schedulers.

Reply:
I didn't get the suggestion completely. Are you suggestion we keep both

`lr_scheduler_configs(self) -> List[LRSchedulerConfig]:`
and `lr_schedulers(self) -> List[Dict[str, Any]]`?

What would be the benefit?

Note that the previous annotation

`lr_schedulers(self) -> List[LRSchedulerConfig]:`

was incorrect

> do we need a way for people to access schedulers directly outside the LightningModule

Yes. Notice how many places I had to rename because of its use outside of the trainer/strategy.

Reply:
It is a pretty simple question. either we answer yes or no. If yes, something like this:

```py
def lr_schedulers(self) -> List[LRScheduler]:
    if version >= 1.8:
        return schedulers
    else: 
        # TODO: remove this warning in 1.8
        warn("Warning: behavior will change in 1.8 ...")
        return configs
```
the schedulers can currently be accessed either in the LM and through filtering them out of the config.

Reply:
Ah I see. I think we can open an issue proposing this change, but this doesn't block this PR.

We would still have a deprecation message, not for the name but for the output.